pkg/action/action_handler.go

```diff
@@ -265,17 +165,61 @@ func getActionTypeFromActionItemDTO(actionItem *proto.ActionItemDTO) (turboactio
 	return actionType, err
 }
 
-// Send action response to Turbonomic server.
-func (handler *ActionHandler) sendActionResult(state proto.ActionResponseState, progress int32, description string) {
-	// 1. build response
-	response := &proto.ActionResponse{
+func (h *ActionHandler) goodResult() *proto.ActionResult {
+
+	state := proto.ActionResponseState_SUCCEEDED
+	progress := int32(100)
+	msg := "Success"
+
+	res := &proto.ActionResponse{
 		ActionResponseState: &state,
 		Progress:            &progress,
-		ResponseDescription: &description,
+		ResponseDescription: &msg,
 	}
-	// 2. built action result.
-	result := &proto.ActionResult{
-		Response: response,
+
+	return &proto.ActionResult{
+		Response: res,
 	}
-	handler.resultChan <- result
+}
+
+func (h *ActionHandler) failedResult(msg string) *proto.ActionResult {
+
+	state := proto.ActionResponseState_FAILED
+	progress := int32(0)
+
+	res := &proto.ActionResponse{
+		ActionResponseState: &state,
+		Progress:            &progress,
+		ResponseDescription: &msg,
+	}
+
+	return &proto.ActionResult{
+		Response: res,
+	}
+}
+
+func keepAlive(tracker sdkprobe.ActionProgressTracker, stop chan struct{}) {
```

Comment:
I still think the keepAlive method should be at SDK level, and the implementation here is definitely not K8s specific.